src/pygeoprocessing/geoprocessing.py

```diff
@@ -2553,6 +2553,8 @@ def convolve_2d(
             "`target_datatype` is set, but `target_nodata` is None. "
             "`target_nodata` must be set if `target_datatype` is not "
             "`gdal.GDT_Float64`.  `target_nodata` is set to None.")
+    if target_nodata is None:
+        target_nodata = float(numpy.finfo(numpy.float32).min)
```

Comment:
Any thoughts on the best way to get this value? Without casting to `float` gdal would raise a type error on `double` when it tried to use this value to. 

Also, I figured using the minimum float64 would be a good idea because we know the output raster will be `GDT_Float64`. And while gdal did _not_ complain about type `double` in that case, it _did_ read the data back in as `-Inf`.

Reply:
@richpsharp , no it works. I just thought there might be a less-verbose way that maybe avoided the need to cast to python's `float`.

And the 2nd question was, why not use the min float64 value since this is a `GDT_Float64` raster? But, I'm happy using the min float32 value, it's certainly small enough.